src/Transactions/TransactionLine.php

```diff
@@ -3,9 +3,39 @@
 namespace PhpTwinfield\Transactions;
 
 use PhpTwinfield\Enums\LineType;
+use PhpTwinfield\ReferenceInterface;
+use PhpTwinfield\Transactions\TransactionFields\LinesField;
 
 interface TransactionLine
 {
+    /**
+     * Get the type of line.
+     *
+     * @return LineType
+     */
     public function getType(): LineType;
+
+    /**
+     * Get the id of the line (or null if not sent to Twinfield yet.
+     *
+     * @return int|null
+     */
     public function getId(): ?int;
+
+    /**
+     * Set the bank transaction on the line. This is needed later on.
+     *
+     * @param LinesField $object
+     */
+    public function setTransaction($object): void;
```

Comment:
Perhaps the visitor pattern is more suited for this job?

Instead of calling `TransactionLine::setTransaction()` on a line whenever it is added to a transaction and exposing the entire transaction, you would call `TransactionLine::visit(LinesField $object)`, which would then scan the transaction for any fields that it needs. This interface could then expose the office and so on retrieved from the transaction.